### ISSUE TITLE
Use Jax primitive instead of masking

### ISSUE THREAD

**Issue by fliingelephant** [fliingelephant/VMC#77]
`jax.lax.dynamic_slice` and `jax.lax.dynamic_update_slice`
